docs/proposals/locking.md

```diff
@@ -108,3 +108,198 @@ support:
+        LatestRemoteCommit string `json:"latest_remote_commit"`
+        // Committer is the individual that wishes to obtain the lock.
+        Committer struct {
+              // Name is the name of the individual who would like to obtain the
+              // lock, for instance: "Rick Olson".
+              Name string `json:"name"`
+              // Email is the email assopsicated with the individual who would
+              // like to obtain the lock, for instance: "rick@github.com".
+              Email string `json:"email"`
+        } `json:"committer"`
+}
+```
+
+```go
+// LockResponse encapsulates the information sent over the API in response to
+// a `LockRequest`.
+type LockResponse struct {
+        // Lock is the Lock that was optionally created in response to the
+        // payload that was sent (see above). If the lock already exists, then
+        // the existing lock is sent in this field instead, and the author of
+        // that lock remains the same, meaning that the client failed to obtain
+        // that lock. An HTTP status of "409 - Conflict" is used here.
+        //
+        // If the lock was unable to be created, this field will hold the
+        // zero-value of Lock and the Err field will provide a more detailed set
+        // of information.
+        //
+        // If an error was experienced in creating this lock, then the
+        // zero-value of Lock should be sent here instead.
+        Lock Lock `json:"lock"`
+        // CommitNeeded holds the minimum commit SHA that client must have to
+        // obtain the lock.
+        CommitNeeded string `json:"commit_needed"`
+        // Err is the optional error that was encountered while trying to create
+        // the above lock.
+        Err error `json:"error,omitempty"`
+}
+```
+
+
+#### `git lfs unlock <path>`
+
+The `unlock` command is responsible for releasing the lock against a particular
+file. The command takes a `<path>` argument which the LFS client will have to
+internally resolve into a UUID to unlock.
+
+The API associated with this command can also be used on the server to remove
+existing locks after a push.
+
+```go
+// An UnlockRequest is sent by the client over the API when they wish to remove
+// a lock associated with the given UUID.
+type UnlockRequest struct {
+        // UUID is the identifier of the lock that the client wishes to remove.
+        UUID string `json:"uuid"`
+}
+```
+
+```go
+// UnlockResult is the result sent back from the API when asked to remove a
+// lock.
+type UnlockResult struct {
+        // Lock is the lock corresponding to the asked-about lock in the
+        // `UnlockPayload` (see above). If no matching lock was found, this
+        // field will take the zero-value of Lock, and Err will be non-nil.
+        Lock Lock `json:"lock"`
+        // Err is an optional field which holds any error that was experienced
+        // while removing the lock.
+        Err error `json:"error,omitempty"`
+}
```

Comment:
Maybe worth saying that the presence of a valid `Lock` and lack of an `Err` implies success. I also assume HTTP status code are going to signal general success/fail.